data/scenarios/Tutorials/require.yaml

```diff
@@ -0,0 +1,79 @@
+name: Require devices
+description: |
+  Learn how to require additional devices that would otherwise not be installed.
+goal:
+    - The 'build' command automatically installs devices that it knows
+      will be required.  For example, if you 'build {move}', some 'treads'
+      will automatically be installed on the new robot since it needs
+      them to 'move'.
+    - However, sometimes you need a device but 'build' can't tell that
+      you need it. In this case, you can use the special 'require'
+      command to require a particular device.  For example, if you
+      'build {require "3D printer"; move}', a 3D printer will be
+      installed on the new robot even though it does not execute any
+      commands that use a 3D printer.
+    - Your goal is to pick a flower on the other side of the river and
+      bring it back to your base.  You win when the base has a flower
+      in its inventory.
+    - "Hint: robots will drown in the water unless they have a 'boat' device
+      installed!"
+entities:
+  - name: periwinkle
+    display:
+      attr: flower
+      char: '*'
+    description:
+    - A flower.
+    properties: [known, infinite, portable]
+  - name: knownwater
+    display:
+      attr: water
+      char: '~'
+    description:
+    - A running stream of of water.
+    properties: [known, portable, growable, liquid]
+    growth: [0,0]
+win: |
+  try {
+    as base {has "periwinkle"}
+  } { return false }
+solution: |
+  build {require "boat"; move; move; move; move; move; f <- grab; turn back; move; move; move; move; move; give base f}
+robots:
+  - name: base
+    loc: [0,0]
+    dir: [1,0]
+    display:
+      char: Ω
+      attr: robot
+    devices:
+      - logger
+      - 3D printer
+    inventory:
+      - [10, solar panel]
+      - [10, logger]
+      - [10, treads]
+      - [10, boat]
+      - [10, grabber]
```

Comment:
> There is no need to have so many devices in inventory because if the first robot grabs the flower and does not give it to the base you have failed.

Actually, that's not true: I made the flower `infinite` for exactly this reason.  However, I think we definitely should offer the ability to start over.

Reply:
Filed #549 to track this.